algorithms/spot_finding/factory.py

```diff
@@ -403,7 +403,7 @@ class SpotFinderFactory:
     """
 
     @staticmethod
-    def from_parameters(params=None, experiments=None):
+    def from_parameters(params=None, experiments=None, is_stills=False):
```

Comment:
`is_stills` -> `nasty_hack_is_stills_flag`

Predicate: this fix is sufficient to fix `dials.stills_process` with data which is in `ImageSequence`s not `ImageSet`s (which would be very useful)

Annotation: adding `is_stills` everywhere is a nasty hack and one which I would like to see reverted once we have resolved the underlying design issues in `dxtbx`. I am however very aware that this will not happen in a timely manner and meanwhile this fix is very well justified. What I would _not_ like to happen is have this stay in the API for the foreseeable future -> being explicit that this is a nasty (if justified) hack seems sensible.